CHANGELOG.md

```diff
@@ -127,6 +127,95 @@ case it elicits some feedback that requires changes.
+about `*Connection` classes in our
+[Architecture Design][architecture-connection] document.
+
+**What are the benefits of `DataConnection`?**
+
+The new API greatly simplifies mocking. Every `Table::Foo(..)` call has an
+associated `DataConnection::Foo(...)` call. This allows you to set expectations
+on the exact values returned by the client call. See
+[Mocking the Cloud Bigtable C++ Client][howto-mock-data-api] for a complete
+example on how to mock the behavior of `Table` with
+`bigtable_mocks::MockDataConnection`.
+
+The new `DataConnection` API offers more consistency across our libraries. It
+also enables the use of some common library features, such as our
+[`UnifiedCredentialsOption`][guac-dox]. Also, any new features will be added to
+the `DataConnection` API first.
+
+**Do I need to update my code?**
+
+No. If the benefits are not appealing enough, you do not need to update your
+code. All code that currently uses `DataClient` will continue to function as
+before. This includes uses of `testing::MockDataClient`.
+
+However, if you are using `testing::MockDataClient` to mock the behavior of
+`Table` in your tests:
+
+1. I am impressed, that is not easy to do.
+2. Be aware that we have announced our intention to remove classes derived from
+   `DataClient` on or around 2023-05. Your tests will break then.
+3. Please consider using `bigtable_mocks::MockDataConnection`. It will greatly
+   simplify your tests.
+
+**How do I update existing `DataClient` code?**
+
+```c++
+namespace gc = ::google::cloud;
+namespace cbt = ::google::cloud::bigtable;
+
+// An option which affects the `DataClient` / `DataConnection`.
+auto options = gc::Options{}.set<gc::GrpcNumChannels>(4);
+
+// Existing code...
+auto data_client = cbt::MakeDataClient("project-id", "instance-id", options);
+auto table = cbt::Table(data_client, "app-profile-id", "table-id",
+                        cbt::LimitedErrorCountRetryPolicy(7));
+
+// Becomes...
+auto connection = cbt::MakeDataConnection(options);
+auto table = cbt::Table(
+    connection, cbt::TableResource("project-id", "instance-id", "table-id"),
+    gc::Options{}
+        .set<cbt::DataRetryPolicyOption>(
+            cbt::DataLimitedErrorCountRetryPolicy(7)->clone())
+        .set<cbt::AppProfileIdOption>("app-profile-id"));
+```
+
+Note the following differences:
+
+1. Resource identifiers are now packaged as a `TableResource` object and passed
+   directly to the `Table`.
+2. The retry and backoff policy types have changed.
+3. The retry, backoff, and idempotency policies are packaged in `Options` as
+   `shared_ptr`s, instead of passed by value as variadic parameters to
+   `Table(..., Policies&&)`.
+4. App profiles are also passed via `Options`.
+
+**What can't the new API do?**
+
+The new [policies][cbt-modern-policies] do not have a
+`Setup(grpc::ClientContext&)` interface. This interface has not been included
+because we believe that setting up the `grpc::ClientContext`...
```

Comment:
Maybe we should move this CHANGELOG entry to a new document, say `bigtable/doc/migrating-from-dataclient.dox` (or `.md`?) and link that document from the CHANGELOG?